Yank/restraints.py

```diff
@@ -1635,12 +1466,51 @@ def restrain_state(self, thermodynamic_state):
         for name, value in self._parameters.items():
             energy_function += '%s = %f; ' % (name, value.value_in_unit_system(unit.md_unit_system))
 
-        # Create the force
-        n_particles = 6  # number of particles involved in restraint: p1 ... p6
-        restraint_force = openmm.CustomCompoundBondForce(n_particles, energy_function)
+        # Create the CV force components
+        force_components = []
+        distance_force = openmm.CustomBondForce('r')
+        distance_force.addBond(self.restrained_receptor_atoms[-1], self.restrained_ligand_atoms[0], [])
+        force_components.append(['boresch_restraint_distance', distance_force])
+        angle_a_force = openmm.CustomAngleForce('theta')
+        angle_a_force.addAngle(self.restrained_receptor_atoms[1],
+                               self.restrained_receptor_atoms[2],
+                               self.restrained_ligand_atoms[0],
+                               [])
+        force_components.append(['boresch_angle_a', angle_a_force])
+        angle_b_force = openmm.CustomAngleForce('theta')
+        angle_b_force.addAngle(self.restrained_receptor_atoms[2],
+                               self.restrained_ligand_atoms[0],
+                               self.restrained_ligand_atoms[1],
+                               [])
+        force_components.append(['boresch_angle_b', angle_b_force])
+        torsion_a_force = openmm.CustomTorsionForce('theta')
+        torsion_a_force.addTorsion(self.restrained_receptor_atoms[0],
+                                   self.restrained_receptor_atoms[1],
+                                   self.restrained_receptor_atoms[2],
+                                   self.restrained_ligand_atoms[0],
+                                   [])
+        force_components.append(['boresch_torsion_a', torsion_a_force])
+        torsion_b_force = openmm.CustomTorsionForce('theta')
+        torsion_b_force.addTorsion(self.restrained_receptor_atoms[1],
+                                   self.restrained_receptor_atoms[2],
+                                   self.restrained_ligand_atoms[0],
+                                   self.restrained_ligand_atoms[1],
+                                   [])
+        force_components.append(['TorsionB', torsion_b_force])
```

Comment:
I think you forgot to change `TorsionB` to `boresch_torsion_b`.

Reply:
Ah, I did, that might explain the NaN's